### PR TITLE
add analytics tracking to onboarding flow

### PR DESCRIPTION
**Pull request by phoebemirman**
part of https://github.com/AugurProject/augur/issues/4856#issuecomment-557226683

### CODE REVIEW

File: packages/augur-ui/src/modules/auth/actions/update-assets.ts [hunk 1]
```diff
@@ -15,21 +11,33 @@ import { AppState } from 'store';
 import { ThunkDispatch, ThunkAction } from 'redux-thunk';
 import { Action } from 'redux';
 import { formatAttoRep } from 'utils/format-number';
+import { track, ADDED_DAI } from 'services/analytics/helpers';
+import { createBigNumber } from 'utils/create-big-number';
 
 export const updateAssets = (
+  possibleDaiIncrease = false,
```
Comment (bthaile) [inline packages/augur-ui/src/modules/auth/actions/update-assets.ts]: possibleDaiIncrease is always false, is this needed?

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 🌠
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/qo0h049**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/modal/containers/modal-buy-dai.ts [hunk 2]
```diff
@@ -23,6 +24,7 @@ const mergeProps = (sP: any, dP: any, oP: any) => ({
   icon: OnboardingPaymentIcon,
   largeHeader: 'Add DAI to your account',
   currentStep: 3,
+  analyticsEvent: BUY_DAI,
```
Comment (bthaile) [inline packages/augur-ui/src/modules/modal/containers/modal-buy-dai.ts]: is this used in the modal?

Comment (phoebemirman) [reply packages/augur-ui/src/modules/modal/containers/modal-buy-dai.ts]: yes. the component uses it

File: packages/augur-ui/src/modules/auth/actions/update-assets.ts [hunk 3]
```diff
@@ -45,6 +53,12 @@ function updateBalances(
     const legacyRep = formatAttoRep(legacyAttoRep).value;
     const dai = amounts[1];
     const eth = amounts[2];
+    if (
```
Comment (bthaile) [inline packages/augur-ui/src/modules/auth/actions/update-assets.ts]: move this logic out of update-assets. update-assets shouldn't care about analytics

File: packages/augur-ui/src/modules/events/actions/log-handlers.ts [hunk 4]
```diff
@@ -170,7 +170,7 @@ export const handleNewBlockLog = (log: Events.NewBlock) => (
   );
   // update assets each block
   if (getState().authStatus.isLogged) {
-    dispatch(updateAssets());
+    dispatch(updateAssets(true));
```
Comment (bthaile) [inline packages/augur-ui/src/modules/events/actions/log-handlers.ts]: this is the only place updateAssets is called the boolean doesn't have any meaning

**Comment by bthaile** [comment]
getting this error

image

File: packages/augur-ui/src/services/analytics/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    analytics.track(eventName, {
+      userAgent: window.navigator.userAgent,
+      ...payload,
+    });
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+// Onboarding event names
```
Comment (andrewdamelio) [inline packages/augur-ui/src/services/analytics/helpers.ts:30]: 👍

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 🥇
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/vv2h32p**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/modal/onboarding.tsx [hunk 6]
```diff
@@ -34,19 +35,25 @@ export const Onboarding = ({
...
-      {linkContent && <LinkContentSection linkContent={linkContent} />}
-    </main>
-    <div>
-      {buttons.length > 0 && <ButtonsRow buttons={buttons} />}
-      {currentStep && <Stepper currentStep={currentStep} maxSteps={4} /> }
+  condensed,
+  analyticsEvent
+}: OnboardingProps) => {
+  useEffect(() => {
+    analyticsEvent && analyticsEvent();
```
Comment (bthaile) [inline packages/augur-ui/src/modules/modal/onboarding.tsx]: this will get called before every render. do we want that?

**Review by bthaile — APPROVED** [review]
lgtm, much cleaner, 👍

**Pull request merged**